### PR TITLE
FEATURE: SETI-401 Zuul: Kill check pipeline when gate pipeline is sta…

### PR DESCRIPTION
**Pull request by janpriessnitz**
…rted meanwhile

### CODE REVIEW

**Comment by michalvanco** [comment]
ok to test

**Comment by michalvanco** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/zuul-opensource-repository-check/1/ : SUCCESS in 7s
- https://ci.intgdc.com/job/zuul-tox-zuul/1/ : FAILURE in 4s

**Comment by janpriessnitz** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/zuul-opensource-repository-check/12/ : SUCCESS in 9s
- https://ci.intgdc.com/job/zuul-tox-zuul/14/ : FAILURE in 7s

**Comment by janpriessnitz** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/zuul-opensource-repository-check/25/ : SUCCESS in 8s
- https://ci.intgdc.com/job/zuul-tox-zuul/31/ : FAILURE in 7m 34s

**Comment by janpriessnitz** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/zuul-opensource-repository-check/27/ : SUCCESS in 0s
- https://ci.intgdc.com/job/zuul-tox-zuul/33/ : SUCCESS in 7m 33s

File: tests/fixtures/layout-abort-pipelines.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      gerrit:
+        - event: patchset-created
+    success:
+      gerrit:
+        verified: 2
+    failure:
+      gerrit:
+        verified: -2
+  - name: gate
+    manager: IndependentPipelineManager
```
Comment (michalvanco) [inline tests/fixtures/layout-abort-pipelines.yaml]: DependentPipelineManager

File: zuul/scheduler.py [hunk 2]
```diff
@@ -1401,6 +1424,13 @@ def removeAbandonedChange(self, change):
             if item.change.equals(change):
                 self.removeItem(item)
 
+    def removeVersionsOfChange(self, change):
```
Comment (michalvanco) [inline zuul/scheduler.py]: check whether existing `removeAbandonedChange` may be used

File: tests/test_scheduler.py [hunk 3]
```diff
@@ -4504,3 +4504,43 @@ def test_success_pattern(self):
         self.assertIn(
             '- docs-draft-test2 https://server/job/docs-draft-test2/1/',
             body[3])
+
+    def test_layout_abort_pipelines(self):
+        "Test if defining abort-pipelines for pipelines with "
+        "manager other than IndependentPipelineManager raises error"
+        self.config.set('zuul', 'layout_config',
+                        'tests/fixtures/layout-bad-abort-pipelines.yaml')
+        self.assertRaises(Exception, self.sched.reconfigure, self.config)
```
Comment (michalvanco) [inline tests/test_scheduler.py]: check specific error

File: tests/fixtures/layout-abort-pipelines.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      gerrit:
+        verified: 2
+    failure:
+      gerrit:
+        verified: -2
+
+projects:
+  - name: org/project
+    check:
+      - project-test1:
```
Comment (michalvanco) [inline tests/fixtures/layout-abort-pipelines.yaml]:
```
+      - project-check1:
 +        - project-check2
 +    gate:
 +      - project-gate
```
for better undestanding in test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/zuul-opensource-repository-check/50/ : SUCCESS in 9s
- https://ci.intgdc.com/job/zuul-tox-zuul-el7/60/ : FAILURE in 6m 57s

**Comment by janpriessnitz** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/zuul-opensource-repository-check/51/ : SUCCESS in 12s
- https://ci.intgdc.com/job/zuul-tox-zuul-el7/61/ : FAILURE in 9m 08s

File: zuul/scheduler.py (hunk 2 above)
Comment (janpriessnitz) [reply zuul/scheduler.py]: There are several existing removeSomething functions. Unfortunately, each of them only removes some special case of the job in the pipeline, so we cannot use them.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/zuul-opensource-repository-check/52/ : SUCCESS in 7s
- https://ci.intgdc.com/job/zuul-tox-zuul-el7/62/ : SUCCESS in 7m 33s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/zuul-opensource-repository-check/53/ : SUCCESS in 10s
- https://ci.intgdc.com/job/zuul-tox-zuul-el7/63/ : FAILURE in 6m 39s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/zuul-opensource-repository-check/54/ : SUCCESS in 8s
- https://ci.intgdc.com/job/zuul-tox-zuul-el7/64/ : FAILURE in 7m 42s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/zuul-opensource-repository-check/56/ : SUCCESS in 7s
- https://ci.intgdc.com/job/zuul-tox-zuul-el7/66/ : SUCCESS in 7m 15s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/zuul-merge-compliance-check/9/ : SUCCESS in 11s
- https://ci.intgdc.com/job/zuul-tox-zuul-el7/67/ : SUCCESS in 8m 20s

**Pull request merged**